src/messaging.js

```diff
@@ -0,0 +1,58 @@
+import {lmsAddress} from './config';
+
+export default class Messaging {
+  constructor(tweet) {
+    this.tweet = tweet;
+  }
+
+  _handleOpenConnection() {
+    console.log('Connection is alive and kicking');
+  }
+
+  _handleMessage(data) {
+    console.log('Received a new message from LMS', data);
+    if (data.topic === 'TWITTER_UPDATE') {
+      this.tweet.update(data.message);
+    }
+  }
+
+  _handleError(error) {
+    console.error('An error happen with connection to LMS', error.stack);
+  }
+
+  _handleEnd() {
+    console.log('Connection closed with LMS');
+  }
+
+  _sendMessage(message) {
+    if (!this.primus) return console.log('There is no connection with LMS');
+    this.primus.write(message);
+  }
+
+  _sendDiconnectionMessage() {
+    this._sendMessage({topic: 'TWITTER_DISCONECT'});
```

Comment:
@rodrigopavezi Wondering where you were planning on using this `disconnect` message?

Reply:
@jungeunyoon that was before our talk yesterday. But I still think we should send a disconnect message so the twitter modules stop the stream for a component id when the component is not being shown on the presentation because it is another item on the play list. Thoughts? cheers

Reply:
@rodrigopavezi you're suggesting that components disconnect when it's not in `play` state, correct? is this necessary? 

two things:

1) with our current architecture it would not only stop the stream but it would also remove the component from the components list on module-side. that's a little drastic and would cause lags since we'd have to make a fresh GET call and open a stream with every turn, don't you think?
2) even if we change the design to only closing the stream on disconnection, i think the components should continue to be updated when they're not being played like with spreadsheet widget, otherwise we risk missing tweets that were made during the pause/stop states.

Reply:
@jungeunyoon fair enough, good points. I just think we can go with it for now but we need to see if that will not decrease the performance of the presentation. Usually we try to stop everything a widget is doing when it is not playing so it does not consume player resource which could crash the browser. That has happened before. I will remove the disconnect message for now. cheers

Reply:
@rodrigopavezi well okay then maybe keep it there for now and we can experiment like you said? 

as stated before, i noticed that spreadsheet widget still gets the updates in the background. it just doesn't reflect changes if it's being played. guess hitting google will take up less resources than connecting to LMS and the other modules. good point.

Reply:
@rodrigopavezi i see you already removed it. hm. well it's your call at this point whether or not to add it back but everything looks good 👍

Reply:
@jungeunyoon yeah, lets see how it goes. If we need it we can come back to it later. cheers

Reply:
@rodrigopavezi sounds good, thanks!